src/safeds_stubgen/api_analyzer/_ast_visitor.py

```diff
@@ -262,10 +270,78 @@ def enter_funcdef(self, node: mp_nodes.FuncDef) -> None:
                 # Sort for the snapshot tests
                 type_var_types.sort(key=lambda x: x.name)
 
+        # Check docstring parameter types vs code parameter type hint
+        for i, argument in enumerate(arguments):
+            arg_type = argument.type
+            doc_type = argument.docstring.type
+
+            if arg_type is None and doc_type is not None:
+                # No code type hint but docstring type
+                arguments[i] = Parameter(
+                    id=argument.id,
+                    name=argument.name,
+                    is_optional=argument.docstring.default_value != "",
+                    default_value=argument.docstring.default_value,
+                    assigned_by=argument.assigned_by,
+                    docstring=argument.docstring,
+                    type=doc_type,
+                )
+            elif arg_type is not None and doc_type is not None and arg_type != doc_type:
+                # Both but different - type hint and doc type
+                if self.type_source_preference == TypeSourcePreference.THROW_WARNING:
+                    # Throw a warning but continue (prefering the code type per default)
```

Comment:
```suggestion
                    # Warn but continue (prefering the code type per default)
```

Reply:
I removed the comment since the option is now split and has nothing to do with the preference.